src/frontend/context/ManuellJournalførContext.tsx

```diff
@@ -0,0 +1,634 @@
+};
+
+const erPersonTom = (person: IPersonInfo | undefined) => !person || !person.personIdent;
+
+const erAvsenderTom = (avsender: AvsenderMottaker | undefined) => !avsender || !avsender.navn;
+
+const patchTomFelt = (ress: Ressurs<IDataForManuellJournalføring>) => {
+    // we use tom object for person and avsender if they are not present in data
+    // because we need to use the objects to index the validation errors (See validaterData() for details)
+    if (ress.status === RessursStatus.SUKSESS) {
+        if (!ress.data.person) {
+            ress.data.person = tomPerson;
+        }
+        if (!ress.data.journalpost.avsenderMottaker) {
+            ress.data.journalpost.avsenderMottaker = tomAvsender;
+        }
+    }
+    return ress;
+};
+
+const validaterData = (dataForValidering: IDataForManuellJournalføring) => {
+    const valideringsfeilMap = new Map<unknown, string[]>();
+
+    const settValideringsfeil = (data: unknown, feil: string) => {
+        valideringsfeilMap.set(data, [...(valideringsfeilMap.get(data) || []), feil]);
+    };
+
+    if (!dataForValidering.journalpost.tittel) {
+        settValideringsfeil(dataForValidering.journalpost, 'Journalpost tittel må ikke være tom');
+    }
+
+    dataForValidering.journalpost.dokumenter?.forEach(dokument => {
+        if (!dokument.tittel) {
+            settValideringsfeil(dokument, 'Dokument tittel må ikke være tom');
+        }
+    });
+
+    if (erPersonTom(dataForValidering.person)) {
+        settValideringsfeil(dataForValidering.person, 'Bruker er ikke satt');
+    }
+
+    if (erAvsenderTom(dataForValidering.journalpost.avsenderMottaker)) {
+        settValideringsfeil(
+            dataForValidering.journalpost.avsenderMottaker,
+            'Avsender er ikke satt'
+        );
+    }
+
+    return valideringsfeilMap;
+};
+
+const [ManuellJournalførProvider, useManuellJournalfør] = createUseContext(() => {
+    const { axiosRequest, innloggetSaksbehandler } = useApp();
+    const [dataForManuellJournalføring, settDataForManuellJournalføring] = React.useState(
+        byggTomRessurs<IDataForManuellJournalføring>()
+    );
+    const [dokumentData, settDokumentData] = React.useState(byggTomRessurs<string>());
+    const { oppgaveId } = useParams<{ oppgaveId: string }>();
+
+    const [valideringsfeil, settValideringsfeil] = React.useState(new Map<unknown, string[]>());
+
+    const harFeil = (data: unknown) => !!hentFeil(data).length;
+
+    const hentFeil = (data: unknown = undefined) =>
+        data
+            ? valideringsfeil.get(data) ?? []
+            : Array.from(valideringsfeil, ([_, feil]) => feil).reduce(
+                  (alleFeil, feil) => [...alleFeil, ...feil],
+                  []
+              );
+
+    const erEndret = () =>
+        oppdatertData.status === RessursStatus.SUKSESS &&
+        dataForManuellJournalføring.status === RessursStatus.SUKSESS &&
+        (JSON.stringify(oppdatertData.data.journalpost) !==
```

Comment:
Not important, but you could also use deep-equal to check equality on objects :)

Reply:
Thank you for letting me know about deepEqual()